enterprise/views.py

```diff
@@ -859,18 +882,25 @@ def post(self, request):
 
         defer_creation = request.POST.get('defer_creation')
         consent_provided = bool(request.POST.get('data_sharing_consent', False))
-        if defer_creation is None and consent_record.consent_required():
-
+        if defer_creation is None and consent_record.consent_required() and consent_provided:
             # Create EnterpriseCourseEnrollment if we found course_run_id instead of course_key in course_id param.
             # Skip creating EnterpriseCourseEnrollment if we found course_key instead of course_run_id.
 
             # EnterpriseCourseEnrollment will be created when the user will select a course run.
             # A CourseEnrollment record will be created and on the post signal of the CourseEnrollment,
             # an EnterpriseCourseEnrollment record will also get created.
-
             if course_id and self.is_course_run_id(course_id):
                 try:
-                    self.create_enterprise_course_enrollment(request, consent_record, course_id)
+                    self.create_enterprise_course_enrollment(request, consent_record, course_id, license_uuid)
+                    if license_uuid:
+                        enrollment_api_client = EnrollmentApiClient()
+                        course_modes = enrollment_api_client.get_course_modes(course_id)
+                        course_mode = 'verified' if 'verified' in course_modes else 'audit'
+                        enrollment_api_client.enroll_user_in_course(
```

Comment:
we might want to catch failures here, if the LMS enrollment fails perhaps we should be redirecting to the failure URL with a notification to try again? also is it possible that this enrollment call is made somewhere else in this flow where we're already handling potential error cases? if so, we may be able to consolidate or simplify.

Reply:
it seems like for other use cases the LMS enrollment happens outside of the DSC flow, so I this should be fine living right here. however the LMS enrollment should happen before the creation of an EnterpriseCourseEnrollment so I'll be switching the order of operations here and redirecting to the failure_url in the case that the LMS enrollment fails